go/vt/vitessdriver/doc.go

```diff
@@ -0,0 +1,113 @@
+// Package vitessdriver contains the Vitess Go SQL driver.
+/*
+Vitess
+
+Vitess is an SQL middleware which turns MySQL/MariaDB into a fast, scalable and
+highly-available distributed database.
+For more information, visit http://www.vitess.io.
+
+
+Example
+
+Using this SQL driver is as simple as:
+
+  import (
+    "time"
+    "github.com/youtube/vitess/go/vt/vitessdriver"
+  )
+
+  func main() {
+    // Connect to vtgate (v1 mode).
+    db, err := vitessdriver.Open("localhost:15991", "keyspace", "0", "master", 30*time.Second)
+
+    // Use "db" via the Golang sql interface.
+  }
+
+For a full example, please see: https://github.com/youtube/vitess/blob/master/examples/local/client.go
+
+The full example is based on our tutorial for running Vitess locally: http://vitess.io/getting-started/local-instance.html
+
+
+Vtgate
+
+This driver will connect to a so called "vtgate" process.
+
+Within Vitess, vtgate is a proxy server which takes care of routing queries to
+the respective shards.
+
+It decouples the routing logic from the application and
+therefore enables Vitess features like consistent, application-transparent
+reshardings:
+When you scale up the number of shards, vtgate will become aware of the new
+shards and you do not have to update your application at all.
+
+Another Vitess feature is the vtgate v3 API. Once enabled, a sharded Vitess
+setup becomes a drop-in replacement for an unsharded MySQL/MariaDB setup.
+
+
+Vtgate API Versions
+
+This driver supports the vtgate API versions v1 and v3, but not v2.
+
+For initial experiments, we recommend to use v1 and then later switch to v3.
```

Comment:
We recommend you use the v1 API for initial experiments and then switch to v3 later.

(You might want to add a sentence explaining why as this seems like it might be a slightly unusual recommendation.)